### PR TITLE
Run EKS CI biweekly and add remove volumes job

### PR DESCRIPTION
**Pull request by misohu**
* Run the eks CI biweekly 
* Remove volume at the end of ekc ci run 
* Set correct env variables in repo 

Successful test run in private repo here https://github.com/misohu/charmed-mlflow-eks-ci/actions/runs/5876696690

### CODE REVIEW

File: .github/workflows/deploy-eks.yaml [hunk 1]
```diff
@@ -85,3 +86,11 @@ jobs:
       - name: Get operator logs
         run: kubectl logs --tail 100 -ntesting -loperator.juju.is/name=mlflow-server
         if: failure()
+
+  delete-unattached-volumes:
+    if: always()
+    uses: ./.github/workflows/delete-aws-volumes.yaml
+    secrets: inherit
+    with:
+      region: eu-central-1
```
Comment (kimwnasptd) [inline .github/workflows/deploy-eks.yaml:95]: Could we define this as an ENV var that gets shared across the whole YAML? I see that other steps are using this as well

File: .github/workflows/deploy-eks.yaml [hunk 2]
```diff
@@ -1,37 +1,38 @@
-name: Configure EKS and run MLflow bundle test
-
+name: Create EKS cluster, deploy CKF and run bundle test
```
Comment (orfeas-k) [inline .github/workflows/deploy-eks.yaml:1]: deploy CKF or MLFlow?

File: .github/workflows/deploy-eks.yaml [hunk 3]
```diff
@@ -1,37 +1,38 @@
...
+      BUNDLE_KUBEFLOW_EKS_AWS_ACCESS_KEY_ID:
         required: true
-      AWS_SECRET_ACCESS_KEY:
+      BUNDLE_KUBEFLOW_EKS_AWS_SECRET_ACCESS_KEY:
         required: true
+  schedule:
+    - cron: "23 0 * * 2"
 jobs:
-  deploy-eks:
+  deploy-ckf-to-eks:
```
Comment (orfeas-k) [inline .github/workflows/deploy-eks.yaml:12]: deploy ckf or mlflow?

File: .github/workflows/deploy-eks.yaml [hunk 4]
```diff
@@ -85,3 +86,11 @@ jobs:
...
         run: kubectl logs --tail 100 -ntesting -loperator.juju.is/name=mlflow-server
         if: failure()
+
+  delete-unattached-volumes:
+    if: always()
+    uses: ./.github/workflows/delete-aws-volumes.yaml
+    secrets: inherit
+    with:
+      region: eu-central-1
+    needs: [deploy-ckf-to-eks]
```
Comment (orfeas-k) [inline .github/workflows/deploy-eks.yaml:96]: is `needs` necessary? won't this always run after the `deploy-ckf-to-eks` action?

**Review by orfeas-k — COMMENTED** [review]
Thank you @misohu, left some minor comments.

File: .github/workflows/deploy-eks.yaml (hunk 1 above)
Comment (misohu) [reply .github/workflows/deploy-eks.yaml:95]: I have nothing against the env, but would you mind if I dont do that  just to stay consistent with the kubeflow-bundle ci ?

Comment (kimwnasptd) [reply .github/workflows/deploy-eks.yaml:95]: @misohu that means that we keep reproducing code we know we don't want, just to stay consistent with code we don't want :)

So no, let's fix this

**Comment by kimwnasptd** [comment]
The changes look good to me, so if @orfeas-k is also OK we should be good to merge this one

**Pull request merged**